validate/validate.h

```diff
@@ -129,6 +130,11 @@ static inline bool IsHostname(const string& to_validate) {
   return true;
 }
 
+static inline size_t Utf8Len(const string& narrow_string) {
+  std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
```

Comment:
@SecurityInsanity This can be implemented without codecvt but counting with protobuf's `UTF8FirstLetterNumBytes` https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/stubs/strutil.h#L881

Reply:
That's good to know. I know @wrowe was working on reworking this for windows, so this may be the easiest solution.

Reply:
Because this is not speed-critical (not any operational aspect of network traffic filtering) the strutil.h implementation seems like an ideal solution, but that implementation is buggy.
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/stubs/strutil.cc#L2284 must read;
3,3,3,3,3,3,3,3, 3,3,3,3,3,3,3,3, 4,4,4,4,4,4,4,4, 5,5,5,5,6,6,1,1
(because no parser in the world will handle the last eight octet values as suggested by this implementation.) Otherwise this is a good choice...

Hitting any 10bbbbbb or 0xfe/0xff byte is considered an unrecognized character and proceeds to advance one byte. Which replicates the go implementation's behavior of allowing unrecognized bytes. The 0xf8-0xfd long characters would be counted, whether they are deciphered or refused by the implementation.